portal-bridge/src/bridge.rs

```diff
@@ -79,9 +79,19 @@ impl Bridge {
     }
 
     async fn launch_test(&self, test_path: PathBuf) {
+        let extension = test_path
+            .extension()
+            .and_then(|x| x.to_str())
+            .expect("Unable to parse test path extension")
+            .to_string();
         let test_asset = fs::read_to_string(test_path).expect("Error reading test asset.");
-        let assets: TestAssets =
-            serde_json::from_str(&test_asset).expect("Unable to parse test asset.");
+
+        let assets: TestAssets = match &extension[..] {
+            "json" => serde_json::from_str(&test_asset).expect("Unable to parse json test asset."),
+            "yaml" => serde_yaml::from_str(&test_asset).expect("Unable to parse yaml test asset."),
+            _ => panic!("Invalid file extension"),
+        };
+
```

Comment:
I would like to extract this block into `utils` because It will be helpful in the beacon network bridge (which will be in a separate file) and maybe the state network bridge.

Reply:
resolved